### PR TITLE
shimv2: use the UnmarshalAny() to unmarshal Protobuf.Any

### PR DESCRIPTION
**Pull request by lifupan**
It'll be much clear to unmarshal Protobuf.Any using
UnmarshalAny().

Fixes: #1130

Signed-off-by: fupan <lifupan@gmail.com>

### CODE REVIEW

File: containerd-shim-v2/exec.go [hunk 1]
```diff
@@ -73,9 +73,11 @@ func newExec(c *container, stdin, stdout, stderr string, terminal bool, jspec *g
 
 	// process exec request
 	var spec specs.Process
-	if err := json.Unmarshal(jspec.Value, &spec); err != nil {
+	v, err := typeurl.UnmarshalAny(jspec)
+	if err != nil {
 		return nil, err
 	}
+	spec = *v.(*specs.Process)
```
Comment (sboeuf) [inline containerd-shim-v2/exec.go]: Please add a switch case to check for the type, and if this is not the expected type, let's throw an error.

File: containerd-shim-v2/service.go [hunk 2]
```diff
@@ -701,11 +701,13 @@ func (s *service) Update(ctx context.Context, r *taskAPI.UpdateTaskRequest) (*pt
 	defer s.Unlock()
 
 	var resources specs.LinuxResources
-	if err := json.Unmarshal(r.Resources.Value, &resources); err != nil {
-		return empty, err
+	v, err := typeurl.UnmarshalAny(r.Resources)
+	if err != nil {
+		return nil, err
 	}
+	resources = *v.(*specs.LinuxResources)
```
Comment (sboeuf) [inline containerd-shim-v2/service.go]: Please add a switch case to check for the type, and if this is not the expected type, let's throw an error.

File: containerd-shim-v2/exec.go (hunk 1 above)
Comment (lifupan) [reply containerd-shim-v2/exec.go]: @sboeuf updated.

File: containerd-shim-v2/service.go (hunk 2 above)
Comment (lifupan) [reply containerd-shim-v2/service.go]: updated.

**Comment by sboeuf** [comment]
@egernst PTAL

**Comment by bergwolf** [comment]
LGTM

**Comment by bergwolf** [comment]
`jenkins-ci-fedora-vsocks` failed on network issue. Restarting...

**Comment by sboeuf** [comment]
`jenkins-ci-fedora-vsocks` does not work at all, as it fails after about 20 seconds. This PR does not change anything related to vsock, so let's move forward and merge it.

**Pull request merged**